#!/usr/bin/env python

PACKAGE = "humanoid_state_estimation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("accb_qx", double_t, 0, " ",   0.003904, 1e-10,   1.0)
gen.add("accb_qy", double_t, 0, " ",   0.003119, 1e-10,   1.0)
gen.add("accb_qz", double_t, 0, " ",   0.004961, 1e-10,   1.0)

gen.add("gyrb_qx", double_t, 0, " ",   0.00523, 1e-10,   1.0)
gen.add("gyrb_qy", double_t, 0, " ",   0.00523, 1e-10,   1.0)
gen.add("gyrb_qz", double_t, 0, " ",   0.00523, 1e-10,   1.0)

gen.add("acc_qx", double_t, 0, " ",   0.0163, 1e-10,   1.0)
gen.add("acc_qy", double_t, 0, " ",   0.0135, 1e-10,   1.0)
gen.add("acc_qz", double_t, 0, " ",   0.0156, 1e-10,   1.0)

gen.add("gyr_qx", double_t, 0, " ",   0.009525, 1e-10,   1.0)
gen.add("gyr_qy", double_t, 0, " ",   0.00641, 1e-10,   1.0)
gen.add("gyr_qz", double_t, 0, " ",   6.5190e-06, 1e-10,   1.0)

gen.add("support_qpx", double_t, 0, " ",   5.0e-03, 1.0e-10,   1.0)
gen.add("support_qpy", double_t, 0, " ",   5.0e-03, 1.0e-10,   1.0)
gen.add("support_qpz", double_t, 0, " ",   5.0e-03, 1.0e-10,   1.0)

gen.add("support_qax", double_t, 0, " ",   5.0e-03, 1.0e-10,   1.0)
gen.add("support_qay", double_t, 0, " ",   5.0e-03, 1.0e-10,   1.0)
gen.add("support_qaz", double_t, 0, " ",   5.0e-03, 1.0e-10,   1.0)

gen.add("support_px", double_t, 0, " ",   5.0e-05, 1.0e-10,   1.0)
gen.add("support_py", double_t, 0, " ",   5.0e-05, 1.0e-10,   1.0)
gen.add("support_pz", double_t, 0, " ",   5.0e-05, 1.0e-10,   1.0)

gen.add("support_ax", double_t, 0, " ",   5.0e-05, 1.0e-10,   1.0)
gen.add("support_ay", double_t, 0, " ",   5.0e-05, 1.0e-10,   1.0)
gen.add("support_az", double_t, 0, " ",   5.0e-05, 1.0e-10,   1.0)

gen.add("odom_px", double_t, 0, " ",   0.0001, 1.0e-10,   1.0)
gen.add("odom_py", double_t, 0, " ",   0.0001, 1.0e-10,   1.0)
gen.add("odom_pz", double_t, 0, " ",   0.0001, 1.0e-10,   1.0)

gen.add("odom_ax", double_t, 0, " ",   0.01, 1.0e-10,   1.0)
gen.add("odom_ay", double_t, 0, " ",   0.01, 1.0e-10,   1.0)
gen.add("odom_az", double_t, 0, " ",   0.01, 1.0e-10,   1.0)

gen.add("com_q", double_t, 0, " ",   1.0e-10, 1.0e-15,   1.0)
gen.add("fd_q", double_t, 0, " ",   5.0, 1.0e-4,   100.0)

gen.add("comd_q", double_t, 0, " ",   1.0e-4, 1.0e-10,   1.0)
gen.add("com_r", double_t, 0, " ",   1.0e-8, 1.0e-10,   1.0)
gen.add("comdd_r", double_t, 0, " ",   5.0e-4, 1.0e-10,   1.0)
exit(gen.generate(PACKAGE, "humanoid_state_estimation", "VarianceControl"))
